### PR TITLE
remove legacy tag events

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Remove the tag events that no longer needs.

### CODE REVIEW

File: Wire-iOS Share Extension/ShareExtensionAnalytics.swift [hunk 1]
```diff
@@ -36,10 +36,6 @@ enum AttachmentType:Int, CaseIterable {
 
 final class ExtensionActivity {
```
Comment (typfel) [inline Wire-iOS Share Extension/ShareExtensionAnalytics.swift:37]: Do we not track anything from the share extension anymore? Can we remove this whole class?

Comment (billypchan) [reply Wire-iOS Share Extension/ShareExtensionAnalytics.swift:37]: checked that this class also stores the attachments info for `appendFileTextToEditor` method and it should stay.

Comment (typfel) [reply Wire-iOS Share Extension/ShareExtensionAnalytics.swift:37]: Seems like this class had multiple responsibilities. We remove all tracking related properties and give the file better name which reflects its responsibility.

The stored attachments are used to calculate `numberOfImages`, `hasVideo`, `hasFile` which are private no longer in use so they can be deleted.

File: Wire-iOS Share Extension/ShareExtensionAnalytics.swift [hunk 2]
```diff
@@ -72,37 +68,6 @@ final class ExtensionActivity {
     func markConversationDidDegrade() {
```
Comment (typfel) [inline Wire-iOS Share Extension/ShareExtensionAnalytics.swift:68]: Can be deleted if `conversationDidDegrade` is deleted.

File: Wire-iOS Share Extension/ShareExtensionAnalytics.swift [hunk 3]
```diff
@@ -36,10 +36,6 @@ enum AttachmentType:Int, CaseIterable {
 
 final class ExtensionActivity {
 
-    static private var openedEventName = "share_extension_opened"
-    static private var sentEventName = "share_extension_sent"
-    static private var cancelledEventName = "share_extension_cancelled"
-
     private var verifiedConversation = false
     private var conversationDidDegrade = false
```
Comment (typfel) [inline Wire-iOS Share Extension/ShareExtensionAnalytics.swift:40]: Private and no longer in use so they can be deleted.

File: Wire-iOS Share Extension/ShareExtensionViewController.swift [hunk 4]
```diff
@@ -294,19 +293,9 @@ final class ShareExtensionViewController: SLComposeServiceViewController {
...
 
-    override func cancel() {
-        if let event = extensionActivity?.cancelledEvent() {
-            sharingSession?.analyticsEventPersistence.add(event)
-        }
-        super.cancel()
-    }
-
     private func storeTrackingData(completion: @escaping () -> Void) {
         extensionActivity?.hasText = !contentText.isEmpty
```
Comment (typfel) [inline Wire-iOS Share Extension/ShareExtensionViewController.swift:297]: Can be deleted, `hasText` was only read by `sentEvent(completion: @escaping (StorableTrackingEvent)`. So I guess this whole function can be deleted.

File: Wire-iOS Share Extension/ShareExtensionViewController.swift [hunk 5]
```diff
@@ -113,7 +113,6 @@ final class ShareExtensionViewController: SLComposeServiceViewController {
         CrashReporter.setupAppCenterIfNeeded()
         updateAccount(currentAccount)
         let activity = ExtensionActivity(attachments: extensionContext?.attachments.sorted)
-        sharingSession?.analyticsEventPersistence.add(activity.openedEvent())
```
Comment (typfel) [inline Wire-iOS Share Extension/ShareExtensionViewController.swift:116]: What about the `analyticsEventPersistence` should we remove this property? Will we add new events to the share extension later?

**Review by typfel — CHANGES_REQUESTED** [review]

File: Wire-iOS Share Extension/ShareExtensionViewController.swift (hunk 5 above)
Comment (billypchan) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:116]: no plannings for tagging share extension events for now.

**Comment by billypchan** [comment]
@typfel I replaced `ExtensionActivity` with a member property `attachments: [AttachmentType: [NSItemProvider]]`

File: Wire-iOS Share Extension/ShareExtensionAnalytics.swift [hunk 6]
```diff
@@ -34,76 +34,13 @@ enum AttachmentType:Int, CaseIterable {
...
-                "text": self.hasText,
-                "conversation_did_degrade": self.conversationDidDegrade
-            ]
-        )
-
-        completion(event)
-    }
-
-}
+//final class ExtensionActivity {
```
Comment (typfel) [inline Wire-iOS Share Extension/ShareExtensionAnalytics.swift:37]: Commented out code

Comment (billypchan) [reply Wire-iOS Share Extension/ShareExtensionAnalytics.swift]: removed.

**Review by typfel — APPROVED** [review]

**Pull request merged**